packages/react-components/src/gallery/dominantSpeaker.test.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { VideoGalleryParticipant } from '..';
+import { smartDominantSpeakerParticipants } from './dominantSpeaker';
+
+const participants: VideoGalleryParticipant[] = [
+  { userId: '1' },
+  { userId: '2' },
+  { userId: '3' },
+  { userId: '4' },
+  { userId: '5' },
+  { userId: '6' },
+  { userId: '7' },
+  { userId: '8' }
+];
+
+const dominantSpeakers: Array<string> = ['1', '3', '5', '7'];
+
+const visibleParticipants: VideoGalleryParticipant[] = [
+  { userId: '1' },
+  { userId: '2' },
+  { userId: '3' },
+  { userId: '4' }
+];
+
+describe('Test smartDominantSpeakerParticipants function', () => {
+  test('returns all participants when number of participants is less than or equal to number of dominant speakers', () => {
+    let result = smartDominantSpeakerParticipants(participants.slice(0, 3), dominantSpeakers, visibleParticipants);
```

Comment:
The tests that don't verify how `visibleParticipants` is handled, shoud pass in an empty list.